index.rst

```diff
@@ -632,15 +641,76 @@ The disassembler function signature is:
             The object that will be disassembled into component parts.
         dataId : dict
             The dataId that is being used to persist this object.
-        componentDict : dict
-            A dict to populate with the components that should be persisted for
-            this composite.
+        componentInfo : dict of componentDatasetName to ComponentInfo
+            A dict of ComponentInfo instances to populate with the components
+            that should be persisted for this composite. Keys are the component
+            names as defined by the datasetType in the policy. Values are
+            ComponentInfo objects.
 
         Returns
         -------
         None
         """
 
+ComponentInfo
+"""""""""""""
+
+ComponentInfo is a class used when assembling and disassembling a composite
+object in butler. It contains information about a component of a composite
+object and has a slot for passing the component object into the assembler and for
+the disssembler to pass the component object out.
+
+ComponentInfo is used as an input to assemblers and disassemblers (which are
+part of the butler public API).
+
+Some ComponentInfo parameters are populated with information from the policy and
+some are filled in by the butler, and when disassembling a composite object the
+obj slot is populated by the disassembler. More details are available in the
+class docstrings.
+
+Generic Assembler & Disassembler
+""""""""""""""""""""""""""""""""
+
+Butler has a generic assembler and disassembler that can be used in some cases,
+listed below. The list is in the order butler will try to apply the generic
```

Comment:
The generic assembler is tried whenever one is not named in the policy (item 1).  It _works_ when any of item 2 or item 3 hold.  It is _not_ required that all of items 1, 2, and 3 simultaneously hold.